src/modules/git_commit.rs

```diff
@@ -80,23 +50,27 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-fn format_tag(symbol: &str, tag_name: &str) -> Option<Result<String, StringFormatterError>> {
-    if tag_name.is_empty() {
-        None
-    } else {
-        Some(Ok(format!("{}{}", &symbol, &tag_name)))
-    }
+fn git_tag(repo: &Repo) -> Option<String> {
+    let git_repo = repo.open();
+    let mut git_head = git_repo.head().ok()?;
+    let head_commit = git_head.peel_to_commit_in_place().ok()?;
+
+    let describe_platform = head_commit.describe().names(AnnotatedTags);
+    let formatter = describe_platform.try_format().ok()??;
+
+    Some(formatter.name?.to_string())
 }
 
-/// len specifies length of hex encoded string
-fn id_to_hex_abbrev(bytes: &[u8], len: usize) -> String {
-    bytes
-        .iter()
-        .map(|b| format!("{:02x}", b))
-        .collect::<String>()
-        .chars()
-        .take(len)
-        .collect()
+fn git_hash(repo: &Repo, config: &GitCommitConfig) -> Option<String> {
+    let git_repo = repo.open();
+    let mut git_head = git_repo.head().ok()?;
+    let head_commit = git_head.peel_to_commit_in_place().ok()?;
+    let head_id = head_commit.id();
```

Comment:
There is a [`head_id()`](https://github.com/Byron/gitoxide/blob/b362fb594546400e6c42688103df438954df7eeb/git-repository/src/repository/reference.rs#L194) method as well.